docs/concepts/time-and-windows.rst

```diff
@@ -0,0 +1,122 @@
+.. _time-and-windows:
+
+Time and Windows
+################
+
+During the life cycle of a record, it experiences a number of events
+which occur at particular times. Three important times in a record's 
+life cycle are *Event-time*, *Ingestion-time*, and *Processing-time*. 
+
+Representing time consistently enables aggregation operations o streams,
+like SUM, that have time boundaries.
+
+KSQL and Kafka Streams support the following notions of time:
+
+Event-time
+    The time when a record is created by the data source. Achieving event-time
+    semantics requires embedding timestamps in records when an event occurs and
+    the record is produced.
+
+    For example, if the event is a geo-location change reported by a GPS sensor
+    in a car, the associated event-time is the time when the GPS sensor captured
+    the location change.
+
+Ingestion-time
+    The time when a record is stored in a topic partition by a Kafka broker.
+    Ingestion-time is similar to event-time, as a timestamp gets embedded in the
+    record, but the ingestion timestamp is generated when the Kafka broker appends
+    the record to the target topic.
+    
+    Ingestion-time may approximate event-time reasonably well if the time
+    difference between creation of the record and its ingestion into Kafka
+    is sufficiently small. This means that ingestion-time may be an alternative
+    for use cases where event-time semantics aren't possible.
+    
+    You may face this situation when data producers don't embed timestamps in
+    records, as in older versions of Kafka’s Java producer client, or when the
+    producer can't assign timestamps directly, like when it doesn't have access
+    to a local clock.
+
+Processing-time
+    The time when the record is consumed by a stream processing application.
+    The processing-time happen immediately after Ingestion-time, or it may
+    happen much later, in milliseconds, hours, or days.
+
+    For example: imagine an analytics application that reads and processes the
+    geo-location data reported from car sensors, and presents it to a
+    fleet-management dashboard. In this case, processing-time in the analytics
+    application might be milliseconds or seconds after event-time, like for
+    real-time pipelines based on Apache Kafka and Kafka Streams) or it might be
+    hours, like for batch pipelines based on Apache Hadoop or Apache Spark.
+
+Kafka Streams assigns a timestamp to every data record by using

```

Comment:
This is kstreams implementation detail.  What does KSQL support?  I think I'm write in thinking its only ingestion or event time, (depending on if the data is written to kafka with the record's timestamp field set to ingestion or event time).

Does KSQL support anything else when it comes to timestamps?  cc @dguy